core/src/test/java/com/google/cloud/sql/core/ConnectorKeyTest.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.core;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.sql.AuthType;
+import com.google.cloud.sql.ConnectionConfig;
+import java.util.Arrays;
+import org.junit.Test;
+
+public class ConnectorKeyTest {
+
+  @Test
+  public void defaultsAreEqual() {
+    ConnectorKey k1 =
+        new ConnectorKey(
+            new ConnectionConfig.Builder().withCloudSqlInstance("proj:name:inst").build());
+    ConnectorKey k2 =
+        new ConnectorKey(
+            new ConnectionConfig.Builder().withCloudSqlInstance("proj:name:inst").build());
+
+    assertThat(k1).isEqualTo(k2);
+    assertThat(k1.hashCode()).isEqualTo(k2.hashCode());
+  }
+
+  @Test
+  public void instanceNameDifferentKeyStillEqual() {
+    ConnectorKey k1 =
+        new ConnectorKey(
+            new ConnectionConfig.Builder().withCloudSqlInstance("proj:name:inst1").build());
+    ConnectorKey k2 =
+        new ConnectorKey(
+            new ConnectionConfig.Builder().withCloudSqlInstance("proj:name:inst2").build());
+
+    assertThat(k1).isEqualTo(k2);
+    assertThat(k1.hashCode()).isEqualTo(k2.hashCode());
+  }
+
+  @Test
+  public void adminUrlNotEqual() {
```

Comment:
Rename to `testKeysNotEqual_withAdminUrlNotEqual` or something similar. Same for the other tests.

Reply:
Fixed.